sycl/doc/extensions/index.rst

```diff
@@ -0,0 +1,54 @@
+=====================================
+DPC++ Extensions to the SYCL Language
+=====================================
+
+The DPC++ compiler supports several extensions to the SYCL language, which are
+divided into several categories.
+
+Supported Extensions
+====================
+
+The "supported" extensions are considered stable.
+Future releases of the compiler may add APIs to these extensions, but existing
+APIs will generally be compatible with new compiler releases.
+If it becomes necessary to drop support for an API, it will first go through a
+deprecation period before being removed.
+
+.. toctree::
+   :maxdepth: 1
+
+   supported/sycl_ext_oneapi_free_function_queries
```

Comment:
Until now, we've directed people to GitHub to read our extensions.  This syntax doesn't render correctly on GitHub, producing this:

![image](https://github.com/user-attachments/assets/fbccbea4-0a8a-4152-849d-894666af759d)

I'm used to finding and reading extensions on GitHub, so I don't really like this.  If we did decide to switch everything over to RST, I think we'd need to make sure we do a good job of highlighting somewhere that people should check the online documentation instead.  I see pros and cons to this: only the stable documentation would be visible on the web, which is good; but it would be harder for people to read through extensions that are implemented in open-source but not yet reflected in the HTML documentation.